src/common/transformations/include/transformations/op_conversions/eye_decomposition.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace pass {
+
+/**
+ * \ingroup ie_transformation_common_api
+ *
+ * \brief Do eye decomposition to sub-graph (model).
+ */
+class TRANSFORMATIONS_API EyeDecomposition : public MatcherPass {
+public:
+    OPENVINO_RTTI("EyeDecomposition", "0");
+    EyeDecomposition();
+
+protected:
+    /**
+     * \brief Make eye model which generate eye matrix.
+     *
+     * If 'k' is outside the eye dimension then result matrix will be filled with zeros.
+     *
+     * \param height  Height of eye
+     * \param width   Width of eye
+     * \param k       Eye diagonal shift.
+     * \param dtype   Data type of eye.
+     *
+     * \return Pointer to decomposed eye model.
+     */
+    std::shared_ptr<Node> make_eye_model(const Output<Node>& height,
```

Comment:
do we really need these functions as class methods?
can we just move them to eye_decomposition.cpp file

Reply:
They are used for test purposes to avoid code duplication to create the expected models.

Reply:
In my opinion, this is not quite the right way to write tests.
Currently, if you make some changes in these methods, the tests will be updated automatically.
So these tests will always pass. Looks like we are testing how the transformation uses this methods, not how to the transformation decomposes Eye operation.

Reply:
Yes I've seen and it just make copy of code what is in transformation. Don't see benefit of doing that as it only give more maintenance. 
The test check if transformation happens then it is the same implementation if not then is failure. 
If you change implementation need modify test code to produce same model and is just duplicate.

Reply:
After discussion with @itikhono added reference transformation in the tests.